static/src/javascripts/projects/common/modules/component.spec.js

```diff
@@ -0,0 +1,418 @@
+
+            if (component.elem) {
+                expect(
+                    component.elem.classList.contains('component--state')
+                ).toBe(false);
+            }
+        });
+
+        test('removeState() should remove class name to elem (with elementName)', () => {
+            component.setState('state', 'element');
+            component.removeState('state', 'element');
+
+            if (subElem) {
+                expect(
+                    subElem.classList.contains('component__element--state')
+                ).toBe(false);
+            }
+        });
+
+        test('toggleState() should toggle class name to elem (without elementName)', () => {
+            component.toggleState('state');
+
+            if (component.elem) {
+                expect(
+                    component.elem.classList.contains('component--state')
+                ).toBe(true);
+            }
+
+            component.toggleState('state');
+
+            if (component.elem) {
+                expect(
+                    component.elem.classList.contains('component--state')
+                ).toBe(false);
+            }
+        });
+
+        test('toggleState() should toggle class name to elem (with elementName)', () => {
+            component.toggleState('state', 'element');
+
+            if (subElem) {
+                expect(
+                    subElem.classList.contains('component__element--state')
+                ).toBe(true);
+            }
+
+            component.toggleState('state', 'element');
+
+            if (subElem) {
+                expect(
+                    subElem.classList.contains('component__element--state')
+                ).toBe(false);
+            }
+        });
+
+        test('hasState() should return the proper state (without elementName)', () => {
+            component.setState('state');
+            expect(component.hasState('state')).toBe(true);
+            expect(component.hasState('whatever')).toBe(false);
+        });
+
+        test('hasState() should return the proper state (with elementName)', () => {
+            component.setState('state', 'element');
+            expect(component.hasState('state', 'element')).toBe(true);
+            expect(component.hasState('whatever', 'element')).toBe(false);
+        });
+
+        test('hasState() should always return false without componentClass', () => {
+            component.componentClass = '';
+
+            component.setState('state', 'element');
+            expect(component.hasState('state')).toBe(false);
+            expect(component.hasState('state', 'element')).toBe(false);
+        });
+    });
+
+    describe('attachTo()', () => {
+        test('calls ready() callback', () => {
+            const component = createComponent({
+                ready: jest.fn(),
+            });
+
+            if (elem) {
+                component.attachTo(elem);
+                expect(component.ready).toBeCalled();
```

Comment:
Shall we go with `.toHaveBeenCalled()`? I know it's an alias, but consistency makes things easier to grok